Challo/Challo/Modules/Trails/TrailListing/TrailListingPresenter.swift

```diff
@@ -42,7 +50,7 @@ class TrailListingPresenter: PresenterProtocol, ObservableObject {
     
     func onPageAppear() {
         isLoading = true
+        self.trails = interactor.getCachedEntities()
         getAllTrails()
     }
```

Comment:
Should `getAllTrails()` be called in `onPageAppear()`? Cmiiw, it should just display what is cached and not have to refetch on each appear?

Reply:
Yea that would be desirable, but if it's the very first time the page is loaded, then the repo is empty and we still need to fetch right. Unless when we are constructing and injecting the repositories we also pre-fetch?

Reply:
Yea i think the latter option sounds good but that gotta wait till kester is done. Maybe can put a TODO `#warning` there first